storage/abort_cache.go

```diff
@@ -0,0 +1,199 @@
+// Copyright 2014 The Cockroach Authors.
```

Comment:
This file and `abort_cache_test` are mostly just code movement with the various bits of cleanup to pare out the unnecessary fields from the original `SequenceCacheEntry`.